src/vis_cpu/vis_gpu.py

```diff
@@ -299,74 +311,366 @@ def vis_gpu(
                     npixc,
                 )
                 events[cc]["eq2top"].record(stream)
-                ## compute tau = dot(antpos,crdtop) / speed_of_light
-                cublas_real_mm(
+
+                tx, ty, tz = crdtop_gpu.get_async(stream=stream)
+                above_horizon = tz > 0
+                tx = tx[above_horizon]
+                ty = ty[above_horizon]
+                nsrcs_up = len(tx)
+                crdtop_lim_gpu = gpuarray.to_gpu_async(
+                    crdtop_gpu.get_async(stream=stream)[:, above_horizon].copy(),
+                    stream=stream,
+                )
+
+                if nsrcs_up < 1:
+                    continue
+
+                tau_gpu = gpuarray.empty(shape=(nant, nsrcs_up), dtype=real_dtype)
+
+                cublas_real_mm(  # compute tau = dot(antpos,crdtop) / speed_of_light
                     h,
                     "n",
                     "n",
-                    npixc,
+                    nsrcs_up,
                     nant,
                     3,
                     ONE_OVER_C,
-                    crdtop_gpu.gpudata,
-                    npixc,
+                    crdtop_lim_gpu.gpudata,
+                    nsrcs_up,
                     antpos_gpu.gpudata,
                     3,
                     0.0,
                     tau_gpu.gpudata,
-                    npixc,
+                    nsrcs_up,
                 )
                 events[cc]["tau"].record(stream)
+
                 ## interpolate bm_tex at specified topocentric coords, store interpolation in A
                 ## threads are parallelized across pixel axis
-                bm_interp(crdtop_gpu, A_gpu, grid=grid, block=block, stream=stream)
-                events[cc]["interpolate"].record(stream)
+                ## Need to do this in polar coordinates or BAD THINGS HAPPEN
```

Comment:
It'd be good to note what bad things happen!

Reply:
👍🏻